src/theory/strings/rewrites

```diff
@@ -87,10 +81,19 @@
 
 (define-rule* str-concat-unify ((s1 String) (s2 String) (s3 String :list) (t2 String) (t3 String :list))
   (= (str.++ s1 s2 s3) (str.++ s1 t2 t3))
-  (= (str.++ s2 s3) (str.++ t2 t3)))
+  (= (str.++ s2 s3) (str.++ t2 t3))
+  _)
```

Comment:
FYI, this was fixed by https://github.com/cvc5/cvc5/commit/32f7d5d1a6c09457f07f632f2f5a398d344b9c54.

I prefer to use the `_` context explicitly since it makes the resulting context clear.